dotnet/src/Connectors/Connectors.Memory.Kusto/KustoMemoryRecord.cs

```diff
@@ -38,7 +38,7 @@ public sealed class KustoMemoryRecord
     /// Initializes a new instance of the <see cref="KustoMemoryRecord"/> class.
     /// </summary>
     /// <param name="record">Instance of <see cref="MemoryRecord"/>.</param>
-    public KustoMemoryRecord(MemoryRecord record) : this(record.Key, record.Metadata, record.Embedding, record.Timestamp) { }
+    public KustoMemoryRecord(MemoryRecord record) : this(record.Key, record.Metadata, record.Embedding, record.Timestamp ?? DateTimeOffset.UtcNow) { }
```

Comment:
I'm not sure if this is the right approach to set `Timestamp` field. As Kusto connector, we should provide an interface to set `Timestamp` field, but we can't decide on behalf of user, what value we should use as default, for some users it can be `UtcNow`, other users may prefer local time based on their scenario. 

Regarding queries like `| where Timestamp > ago(1h)`, in this case I think field `Timestamp` should be as part of Kusto system fields, which should be set automatically on Kusto side.

But I do agree, that `SemanticTextMemory` should accept `timestamp` field as part of schema, because otherwise there is no option to provide this field, unless using `KustoMemoryStore` directly, which is not preferred way in context of Semantic Kernel.

Reply:
Thanks @dmytrostruk , if SemanticTextMemory provides parameter to set this value, can be another possible way, but not all memory store from connectors can handle timestamp correctly. 

For example, in original Kusto MemoryStore code, [here](https://github.com/microsoft/semantic-kernel/blob/cfebcaec6c1656f2c129ca3e40442c0ace9495e5/dotnet/src/Connectors/Connectors.Memory.Kusto/KustoMemoryStore.cs#L134) and [here](https://github.com/microsoft/semantic-kernel/blob/cfebcaec6c1656f2c129ca3e40442c0ace9495e5/dotnet/src/Connectors/Connectors.Memory.Kusto/KustoMemoryStore.cs#L217) , it tries to read timestamp as string, but indeed this field is DateTimeOffset. 

Another thing is Kusto doesn't set this field automatically, that's why I have to set it in Kusto Connector in this way. 

From my point of view, before all connectors can well accept timestamp setting, and SemanticTextMemory starts well support this option, for Kusto Connector scenario, this method is an achievable solution. 

We define the default behavior is UTC DateTimeOffset. If SemanticTextMemory provides this value. Kusto can still use it because the code accept explicitly setting the value from record (which is from SemanticTextMemory(: record.Timestamp ?? DateTimeOffset.UtcNow

Reply:
> but not all memory store from connectors can handle timestamp correctly

`Timestamp` parameter can be optional, as well as other parameters that currently exist. 

> it tries to read timestamp as string, but indeed this field is DateTimeOffset

I think there was some issue with `reader.GetDateTime(2)`, we will need to test it manually and double-check.

> We define the default behavior is UTC DateTimeOffset

As I said in previous comment, I'm not sure that we can/should define this default behavior. In my opinion, it should be `null` by default (as optional field), and users can define their default values as soon as we extend `SemanticTextMemory`. The change in this PR can fix some scenarios for now, but we need to think about long-term solution, which will bring benefits not only for Kusto connector, but other connectors as well.